### DISCUSSION TITLE
Where is the DNS cache stored

### DISCUSSION THREAD

**Discussion by GeniusBroccoli** [DNSCrypt/dnscrypt-proxy#2280] [Q&A] [answered]
Maybe this is a dumb question, but I did not find a clear explanation among other users comments, but where is the DNS cache stored, how to clear it, and are there any diagnostic commands to get information about the cache?

**Comment by jedisct1** [comment, marked as answer]
DNS records are small and ephemeral, so they are cached in memory.

Restarting the proxy will clear it.

And when query logging is enabled, you can see for every query whether the response came directly from the cache, or if it required a request to an upstream server.

**Reply by GeniusBroccoli** [reply]
Thank you very much for the explanation. Is there any point in creating a local cache with additional software?

**Reply by Aadam-Marshall** [reply]
> DNS records are small and ephemeral

Doesn't whether they are ephemeral depend on cache's TTL? What if I set some high value, for example 1 week for `cache_min_ttl`? I would also love to see the cache by the way.
